packages/cashier-v2/src/routes/RouteWithSubRoutes.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Redirect, Route, useLocation } from 'react-router-dom';
+import { TRouteTypes } from '../types';
+import { defaultRoute } from './Router';
+
+const RouteWithSubRoutes = (route: TRouteTypes.IRouteConfig) => {
+    const location = useLocation();
+    const pathname = location.pathname.replace(/\/$/, '');
+
+    return (
+        <Route path={route.path}>
+            {/* Redirection to default route "/cashier-v2/deposit" from "/cashier-v2" */}
+            {pathname === route.path && <Redirect to={defaultRoute?.path} />}
```

Comment:
if I understand correctly - this is redirect only to support /cashier-v2 => /cashier-v2/deposit redirection (so one redirection), but given RouteWithSubroutes is used in a loop within Route, that <Redirect> is gonna be added multiple times if I have read the code correctly

Reply:
maybe worth inlining so this type of quirks going to be immediately visible?

Reply:
hey @wojciech-deriv 

> but given RouteWithSubroutes is used in a loop within Route
You are right

> that is gonna be added multiple times if I have read the code correctly

but this statement is not right actually because if we check the structure of the route config. we are mapping only through one element in array ("/cashier-v2"). that mean that this code will present only once

Reply:
@wojciech-deriv 
eventually i refactored the code for better readability
so now we are not mapping but just using the first element from the config and passing props of that element to `RouteWithSubroutes` component

Reply:
sorted